### PR TITLE
en-US fallback on 404

### LINKED ISSUE #1340: English fallback
Take any English page and manually edit the URL by replacing the `en-US` with something that doesn't have a translation. 
E.g. https://developer.mozilla.org/sv-SE/docs/Web/JavaScript/Reference/Global_Objects/WeakMap
in Kuma, it applies the `sv-SE` chrome localization on the header and footer and stuff. But in terms of the content it does something like this:

```python
try:
  doc = Document.objects.get(slug=slug, locale='sv-Se')
except Document.DoesNotExist:
  try:
    doc = Document.objects.get(slug=slug, locale='en-US')
  except Document.DoesNotExist:
    raise Http404(slug)
```

We can't do that anymore in Yari. We need to either throw a 404 immediately or redirect to the en-US version.

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1340 

Ideally, people shouldn't end up on these URLs anyway, but if they do, let's make the most of it. 
To test, you can just take any URL, and replace `en-US` for something else. Like http://localhost:5000/xxx/docs/MDN/Kitchensink
A XHR request will then try to `GET` the `/en-US/docs/$slug/index.json` and if that XHR request works it presents something. 

Here's what it looks like:
Screen Shot 2020-11-18 at 8 31 59 PM

If anything goes wrong with the XHR request (i.e. not 200 and not 404), this happens:
Screen Shot 2020-11-18 at 8 32 53 PM

And if the whole thing "failed" in that it doesn't exist in `en-US` either, then nothing happens.
Screen Shot 2020-11-18 at 8 42 43 PM

@schalkneethling Out of desperation, I (ab)used the `.notecard` to get some styling. It would be nice if the success note card was green instead of blue. 
Also, it might just be bad to use `.notecard` because this isn't actually a notecard. But like I said; desperation. Ideas?

### CODE REVIEW

**Comment by peterbe** [comment]
The reason this PR is a draft is that its branch was started from [this not-yet-merged PR](https://github.com/mdn/yari/pull/1645).

**Comment by peterbe** [comment]
@schalkneethling Please go ahead and review it and make fun my of scss finger-painting. See what I wrote above about the `.notecard` hacks.

**Review by schalkneethling — COMMENTED** [review]
> If anything goes wrong with the XHR request (i.e. not 200 and not 404), this happens:

The reason the icon sits there all by its lonesome is that the card does not have a heading. If you add something like:

```
<h4>Oh noes!</h4>
```

You will get the following:

Screenshot showing notecard with a heading

**Review by schalkneethling — CHANGES_REQUESTED** [review]
A couple of suggestions, ideas, and changes. Thanks, @peterbe

**Comment by schalkneethling** [comment]
> Also, it might just be bad to use `.notecard` because this isn't actually a notecard. But like I said; desperation. Ideas?

I think that is actually totally fine to use notecards for this. What _I_ do need to do as add a success variant.

**Comment by schalkneethling** [comment]
As mentioned on Slack, as of https://github.com/mdn/yari/pull/1793 you can now use `notecard success`

File: client/src/page-not-found/index.tsx [hunk 1]
```diff
@@ -42,10 +51,95 @@ export function PageNotFound() {
...
+            <small>{document.mdn_url}</small>
+          </Link>
+        </p>
+      </div>
+    );
+  } else if (document === null) {
+    // It means the lookup "worked" in principle, but there wasn't an English
+    // document there. Bummer. But at least we tried.
+    // Should we say something??
+  }
```
[the remark this answers is not in the document]
Comment (peterbe) [reply client/src/page-not-found/index.tsx:142]: The problem with that is that there's probably a better strategy that day. I.e. that idea I had where we download all the URLs in the search-index.json and compare each and every URL as a possible typo.

**Comment by peterbe** [comment]
I'm guessing this'll look nice and green once minimalist upgrades next time:
Screen Shot 2020-11-20 at 11 01 26 AM

**Comment by peterbe** [comment]
And if it goes wrong.
Screen Shot 2020-11-20 at 11 05 28 AM

**Review by schalkneethling — APPROVED** [review]
🦃

File: client/src/page-not-found/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ryuno-Ki) [reply client/src/page-not-found/index.tsx:143]: e.g. comparing using Levenshtein distance …

**Pull request merged**